File: ios/FestivalConnection/Models/PeerInfo.swift
import Foundation

struct PeerInfo: Identifiable, Equatable {
    var id: String { publicKeyHex }
    let publicKeyHex: String
    var displayName: String
    var handle: String
    var lastSeen: Date
    var profileImageData: Data?
    var isConnected: Bool = false
    var isReachable: Bool = false
    var noisePublicKey: Data?

    var connectionQuality: String {
        if isConnected { return "Nearby" }
        if isReachable { return "In range" }
        return "Searching..."
    }
}

File: android/app/src/main/java/com/appxstudios/festivalconnection/models/PeerInfo.kt
package com.appxstudios.festivalconnection.models

data class PeerInfo(
    val publicKeyHex: String,
    var displayName: String,
    var handle: String,
    var lastSeen: Long = System.currentTimeMillis(),
    var profileImageData: ByteArray? = null,
    var isConnected: Boolean = false,
    var isReachable: Boolean = false,
    var noisePublicKey: ByteArray? = null
) {
    val connectionQuality: String
        get() = when {
            isConnected -> "Nearby"
            isReachable -> "In range"
            else -> "Searching..."
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PeerInfo) return false
        return publicKeyHex == other.publicKeyHex
    }

    override fun hashCode(): Int = publicKeyHex.hashCode()
}
